src/ui/ui_nav_link_collection.js

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
+import UiNavLink from './ui_nav_link';
+import Collection from '../utils/Collection';
+
+let inOrderCache = Symbol('inOrder');
+
+module.exports = class UiNavLinkCollection extends Collection {
```

Comment:
Can you switch this to `export default`? Might a well use es6 throughout this whole file, and you're using es6 exports in other files in this PR.